resources/environments/resource_proxy_configuration.go

```diff
@@ -0,0 +1,211 @@
+		tflog.Error(ctx, "Got Error while trying to get plan")
+		return
+	}
+
+	client := p.client.Environments
+
+	params := operations.NewCreateProxyConfigParamsWithContext(ctx)
+	params.WithInput(&models.CreateProxyConfigRequest{
+		ProxyConfigName: data.Name.ValueStringPointer(),
+		Description:     data.Description.ValueString(),
+		Protocol:        data.Protocol.ValueStringPointer(),
+		Host:            data.Host.ValueStringPointer(),
+		Port:            func(i int32) *int32 { return &i }(int32(data.Port.ValueInt64())),
+		NoProxyHosts:    data.NoProxyHosts.ValueString(),
+		User:            data.User.ValueString(),
+		Password:        data.Password.ValueString(),
+	})
+
+	responseOk, err := client.Operations.CreateProxyConfig(params)
+	if err != nil {
+		msg := err.Error()
+		if d, ok := err.(utils.EnvironmentErrorPayload); ok && d.GetPayload() != nil {
+			msg = d.GetPayload().Message
+		}
+
+		resp.Diagnostics.AddError(
+			"Create Proxy Configuration",
+			"Failed to create proxy configuration, unexpected error: "+msg,
+		)
+		return
+	}
+
+	proxyConfig := responseOk.Payload.ProxyConfig
+	data.ID = types.StringValue(*proxyConfig.Crn)
+
+	diags = resp.State.Set(ctx, data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (p *proxyConfigurationResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state proxyConfigurationResourceModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		tflog.Error(ctx, "Got Error while trying to get state")
+		return
+	}
+
+	client := p.client.Environments
+
+	params := operations.NewDeleteProxyConfigParamsWithContext(ctx)
+	params.WithInput(&models.DeleteProxyConfigRequest{ProxyConfigName: state.Name.ValueStringPointer()})
+
+	_, err := client.Operations.DeleteProxyConfig(params)
+	if err != nil {
+		if envErr, ok := err.(*operations.DeleteProxyConfigDefault); ok {
+			if cdp.IsEnvironmentsError(envErr.GetPayload(), "NOT_FOUND", "") {
+				return
+			} else {
+				resp.Diagnostics.AddError(
+					"Delete Proxy Configuration",
+					"Failed to delete proxy configuration: "+envErr.Payload.Message,
+				)
+				return
+			}
+		}
+		msg := err.Error()
+		if d, ok := err.(utils.EnvironmentErrorPayload); ok && d.GetPayload() != nil {
+			msg = d.GetPayload().Message
+		}
+
+		resp.Diagnostics.AddError(
+			"Delete Proxy Configuration",
+			"Failed to delete proxy configuration, unexpected error: "+msg,
+		)
+	}
+}
+
+func (p *proxyConfigurationResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state proxyConfigurationResourceModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		tflog.Error(ctx, "Got Error while trying to get state")
+		return
+	}
+
+	client := p.client.Environments
+
+	params := operations.NewListProxyConfigsParamsWithContext(ctx)
```

Comment:
Extracting line 149-176 (and also 183) into a `FindProxyConfifgurationByName()` method will enable us to use it in acceptance tests later on when they are added. I think as a general pattern, we should write `FindXXXByName` or `FindXXXById` type of methods for every resource / data source to keep the code clean for the `Read` methods. Since we do almost the same thing for every "List" calls for resources, when we add templated code generation, we will also auto-generate those methods.